test/inspector/test-async-hook-setup-at-inspect.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+const common = require('../common');
+common.skipIfInspectorDisabled();
+common.skipIf32Bits();
+common.crashOnUnhandledRejection();
+const { NodeInstance } = require('../inspector/inspector-helper.js');
+const assert = require('assert');
+
+// Even with --inspect, the default async call stack depth is 0. We need a
+// chance to call Debugger.setAsyncCallStackDepth *before* activating the timer
+// for async stack traces to work.
```

Comment:
IIUC, this actually means that even though we enable inspector async hook reporting AsyncTask events to V8 Inspector at startup, the async stack traces are effectively disabled until the debugger connects and increases `m_maxAsyncCallStackDepth`.

I think this make the user experience little bit less worse, because async workflows started before the debugger has connected will not be captured by V8 Inspector.

It makes me wonder if there is a way how to increase `m_maxAsyncCallStackDepth` at the time when the async hook reporting AsyncTasks is enabled? That way the two problematic tests `setup-at-inspect` and `setup-at-signal` can be simplified back to the original form.

Anyways, I think this it will be better to land this pull request as it is, because even if we don't capture async stack traces from the time before the debugger client was connected, it's still better to have some async stack traces with the current code in this patch, rather than have no async stack traces as is the current status in `master`.